datalad/metadata/metadata.py

```diff
@@ -70,6 +76,14 @@
 from datalad.log import log_progress
 from datalad.core.local.status import get_paths_by_ds
 
+# Check availability of new-generation metadata
+try:
+    from datalad_metalad.dump import Dump
+    next_generation_metadata_available = True
+except ImportError:
+    next_generation_metadata_available = False
```

Comment:
I think it would not hurt if we start "versioning" the metadata approach ;) I would count this as `_gen4`

- 1st gen: `.datalad/meta` (old implementation within `datalad`)
- 2nd gen: `.datalad/metadata/aggregate_v1.json` (currently in `datalad`)
- 3rd gen: the "old" one of metalad
- 4th gen: this "next_generation" - current as in `metalad`, right?
- 
May be even add that info into `docs/source/metadata.rst` to reach clarity? WDYT @mih -- I think having some codenames for metadata approaches would be helpful, at least to me.

Reply:
Good point. `_gen4` would be appropriate, if we discriminate between `_gen2` and `_gen3` based on the interface, since the storage is the same (at least in `_gen2` and `_gen3`).

I will start using "generation four".